src/components/swap/index.tsx

```diff
@@ -248,7 +247,7 @@ export const Swap = (props: SwapProps) => {
     }
 
     if (buttonState === TradeButtonState.signTerms) {
-      onOpenSignTermsModal()
+      await signTermsOfService()
```

Comment:
probably want a `return` just to be sure

Reply:
good call, will add 👍